Update digital object metadata display, refs #13388

replaceafill opened the pull request:
This updates the `Digital object metadata` section of the index views with two collapsible fieldsets that look like this:

![Screenshot_2020-08-27 Beihai, Guanxi, China, 1988 - Demo site](https://user-images.githubusercontent.com/560781/91482833-efc08780-e863-11ea-9cfa-9a3139b316f4.png)

The rest of the changes include:

* New properties have been added to information objects to store METS information about the original and preservation files.
* New visibility settings have been added to control what fields are displayed in the new areas.
* A new preservation system access statement setting has been added (similar to the copyright statement) to show a message to unauthenticated users on how to get access to the preservation copies

mcantelon commented on apps/qubit/modules/digitalobject/actions/metadataComponent.class.php:
```diff
@@ -28,34 +28,333 @@ class DigitalObjectMetadataComponent extends sfComponent
+
+  protected function isPreservationSystemAccessStatementEnabled()
+  {
+    if (null !== $setting = QubitSetting::getByName('digitalobject_preservation_system_access_statement_enabled'))
```
Would it work to use `sfConfig::get` to get the property instead?

mcantelon commented on apps/qubit/modules/digitalobject/actions/metadataComponent.class.php:
```diff
@@ -28,34 +28,333 @@ class DigitalObjectMetadataComponent extends sfComponent
+  {
+    if (
+      $this->isPreservationSystemAccessStatementEnabled()
+      && (null !== $setting = QubitSetting::getByName('digitalobject_preservation_system_access_statement'))
```
`sfConfig::get` instead?

mcantelon commented on apps/qubit/modules/digitalobject/actions/metadataComponent.class.php:
```diff
@@ -28,34 +28,333 @@ class DigitalObjectMetadataComponent extends sfComponent
+    }
+  }
+
+  protected function isEmpty($value)
```
Is this being used instead of `empty()` because `empty()` would report `0` as empty?

replaceafill replied:
Yes. Is there a better approach?

mcantelon replied:
No, your approach seems correct given that!

mcantelon commented on apps/qubit/modules/digitalobject/actions/metadataComponent.class.php:
```diff
@@ -28,34 +28,333 @@ class DigitalObjectMetadataComponent extends sfComponent
-    $this->relatedToIo = $this->resource->object instanceOf QubitInformationObject;
-    $this->relatedToActor = $this->resource->object instanceOf QubitActor;
+    $this->showMasterFileGoogleMap = (
+      (bool)sfConfig::get('app_toggleDigitalObjectMap')
```
Is the casting of the value as bool to deal with the setting being null? Seems like we normally use this way of dealing with null values: `sfConfig::get('app_toggleDigitalObjectMap', false)`.

mcantelon commented on apps/qubit/modules/digitalobject/actions/metadataComponent.class.php:
```diff
@@ -28,34 +28,333 @@ class DigitalObjectMetadataComponent extends sfComponent
+    );
+
+    $this->showMasterFileGeolocation = (
+      (bool)check_field_visibility('app_element_visibility_digital_object_geolocation')
```
`check_field_visibility` should, I think, always return a bool so not sure if we need to cast it as bool.

mcantelon commented on lib/QubitMetsParser.class.php:
```diff
@@ -1166,4 +1166,217 @@ public function getOriginalPathInAip($fileId)
+  {
+    if (
+      (false !== $file = $this->document->xpath(
+        '//m:fileSec/m:fileGrp[@USE="original"]/m:file[@ID="'.$fileId.'"]'
```
Nitpick: using sprintf instead of string concatenation's a bit cleaner, but not a big deal either way.

mcantelon reviewed (commented):
Nicely done! A few questions/concerns, but nothing major.

replaceafill commented:
Thanks for the code review and your recommendations @mcantelon! :bow: I think I addressed all your observations.

mcantelon reviewed (approved):
Great work... looks good! 👍

replaceafill committed:
Update digital object metadata display, refs #13388

- Two collapsible fieldsets have been added to the `Digital object
  metadata` section: Preservation Copies and Access Copies.
- New properties have been added to information objects to store METS
  information about the original and preservation files.
- New visibility settings have been added to control what fields are
  displayed in the new areas.
- A new preservation system access statement setting has been added
  (similar to the copyright statement) to show a message to
  unauthenticated users on how to get access to the preservation
  copies.
